beets/dbcore/query.py

```diff
@@ -447,6 +447,34 @@ def match(self, item):
         return any([q.match(item) for q in self.subqueries])
 
 
+class NotQuery(Query):
+    """A query that matches the negation of its `subquery`, as a shorcut for
+    performing `not(subquery)` without using regular expressions."""
+    def __init__(self, subquery):
+        self.subquery = subquery
+
+    def clause(self):
+        clause, subvals = self.subquery.clause()
+        if clause:
+            return 'not ({0})'.format(clause), subvals
+        else:
+            # special case for RegexpQuery, (None, ())
```

Comment:
After playing a bit with the suggestion and inspecting the code more carefully, I'm wondering if the suggested change for explicitness would actually decrease readability? In particular:
* only the `FieldQuery` subclasses have the `fast` attribute, and ideally `NotQuery` should work fine with any 'Query' (for example, `-foo` results in a `NotQuery(AnyQuery(...))`).
* `RegexpQuery` is still a bit problematic, as it is the only `FieldQuery` descendant that does not implement `col_clause()`. As a result, I can't seem to find a way to avoid the explicit checking for `clause != None` at 458-462, otherwise `NotQuery(RegexpQuery(..., fast=True).clause()`  ends up returning `(u'not (None)', ())`, which breaks many test cases.